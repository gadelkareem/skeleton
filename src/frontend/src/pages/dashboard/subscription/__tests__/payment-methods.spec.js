import { mount } from '@vue/test-utils'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import initData from '@@/globals/init-data'
import Vuetify from 'vuetify'
import Page from '../payment-methods'

const vuetify = new Vuetify()

describe('payment-methods.vue', () => {
  let store

  beforeEach(() => {
    const actions = {
      'page/title': jest.fn(),
      'loading/start': jest.fn(),
      'loading/finish': jest.fn()
    }
    const getters = {
      'user/user': () => initData().user
    }
    store = new Vuex.Store({
      actions,
      getters,
      state: {
        loading: { status: false }
      }
    })
    const script = document.createElement('script')
    document.body.appendChild(script)
  })

  it('should match snapshot', () => {
    const router = new VueRouter({})
    const w = mount(Page, {
      store,
      router,
      vuetify,
      stubs: ['router-link', 'router-view'],
      created () {
        this.$vuetify.lang = {
          t: () => {}
        }
        this.$vuetify.breakpoint = {
          width: 0,
          height: 0
        }
        this.$vuetify.theme = { dark: false }
      }
    })

    expect(w.html()).toMatchSnapshot()
    w.destroy()
  })
})
