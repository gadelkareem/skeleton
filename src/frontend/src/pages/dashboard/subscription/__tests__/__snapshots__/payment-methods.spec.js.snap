// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payment-methods.vue should match snapshot 1`] = `
<div class="container"><span> <!----> <!----></span>
  <div class="row">
    <div>
      <div class="container">
        <div class="v-item-group theme--light">
          <div class="container">
            <div class="row">
              <div class="float-left ma-2">
                <div role="dialog" class="v-dialog__container"><button type="button" class="mt-7 v-btn v-btn--is-elevated v-btn--has-bg theme--dark v-size--x-large white" style="height: 150px; width: 300px;" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-plus theme--dark green--text" style="font-size: 40px;"></i> Add Credit Card
    </span></button>
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
